clients/go/test/api_uploads_test.go

```diff
@@ -17,34 +17,52 @@ import (
 	"testing"
 
 	"github.com/antihax/optional"
-	"github.com/phrase/phrase-go"
+	"github.com/phrase/phrase-go/v2"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 func Test_phrase_UploadsApiService(t *testing.T) {
-	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// Send the mock response
-		response := `{"id": "1", "filename": "test.json", "format": "json", "state": "valid" }`
-		w.Header().Set("Content-Type", "application/json")
-		w.WriteHeader(http.StatusOK)
-
-		w.Write([]byte(response))
-	}))
-
-	defer server.Close()
-
-	configuration := phrase.NewConfiguration()
-	configuration.BasePath = server.URL
-	apiClient := phrase.NewAPIClient(configuration)
-
 	t.Run("Test UploadsApiService UploadCreate", func(t *testing.T) {
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// assert request body for proper serialization of tested format_options
+			r.ParseMultipartForm(0)
+			assert.Equal(t, r.FormValue("format_options[translation_columns][en]"), "f")
+			assert.Equal(t, r.FormValue("format_options[translation_columns][de]"), "e")
+			assert.Equal(t, r.FormValue("format_options[top_level_key]"), "100")
+
+			// Send the mock response
+			response := `{"id": "1", "filename": "test.json", "format": "json", "state": "valid" }`
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+	
+			w.Write([]byte(response))
+		}))
+	
+		defer server.Close()
+	
+		configuration := phrase.NewConfiguration()
+		configuration.BasePath = server.URL
+		apiClient := phrase.NewAPIClient(configuration)
+
 		file, _ := os.Create("testfile.json")
 		fileFormat := optional.NewString("json")
 		fileObject := optional.NewInterface(file)
 		localeId := optional.NewString("99")
 
-		localVarOptionals := phrase.UploadCreateOpts{FileFormat: fileFormat, File: fileObject, LocaleId: localeId }
+		// setting format_options
+		formatOptions := make(map[string]interface{})
+		nestedOptions := make(map[string]interface{})
+
+		nestedOptions["en"] = "f"
+		nestedOptions["de"] = "e"
+
+		formatOptions["top_level_key"] = 100
+		formatOptions["translation_columns"] = nestedOptions
+
+		formatOptionsMap := optional.NewInterface(formatOptions)
+
+		localVarOptionals := phrase.UploadCreateOpts{FileFormat: fileFormat, File: fileObject, LocaleId: localeId, FormatOptions: formatOptionsMap }
 		resp, httpRes, err := apiClient.UploadsApi.UploadCreate(context.Background(), "project_id", &localVarOptionals)
 		requestUrl := httpRes.Request.URL
```

Comment:
I think the request content can be also asserted here instead of in the mock server handler. could be cleaner that way, rather having assertions all over the place.

Reply:
Very valid point, actually by the point request is sent, the request's body is no longer available, quoting go's doc

> // Request is the request that was sent to obtain this Response.
	// Request's Body is nil (having already been consumed).
	// This is only populated for Client requests.
	Request *Request

Reply:
Let me still check if there is a better way to achieve this.

Reply:
I tried to look into a more readable alternative. There is another option to use [responseRecorder](https://pkg.go.dev/net/http/httptest#ResponseRecorder) but looking at some examples I think its a similar approach where assertions will be made inside the handler function that responsible to stub the request.

Reply:
it's fine as is, no worries.